03-routing-and-input.Rmd

```diff
@@ -323,4 +324,28 @@ You can print out the names of all of the properties attached to the request by
 
 // TODO: Conflicts between path, query string, body?
 
+### Form submit
+
+One common way of developing API's is to submit a form with its inputs using a POST request. For example to handle a login form we would need: a form to send to the server such as:
+
+```html
+<h2>Basic login form</h2>
+
+<pre>
+    <form action="/login">
+      <input type="text" placeholder="Enter Username" name="uname" required>
+      <input type="password" placeholder="Enter Password" name="psw" required>
```

Comment:
Let's use a different example than username/password since that brings up a number of security considerations. Also some browsers will now render scary warnings if you try to enter a password on non-HTTPS.

Maybe we could do email address and opt-in or something trivial: favorite color text input and a checkbox for something.

Reply:
Done.